src/frontend/src/flows/authenticate/index.ts

```diff
@@ -322,11 +255,79 @@ async function handleAuthSuccess(
   return {
     userNumber: loginResult.userNumber,
     connection: loginResult.connection,
-    sendDelegationMessage: () =>
-      authContext.postMessageCallback({
-        kind: "authorize-client-success",
-        delegations: [parsed_signed_delegation],
-        userPublicKey: Uint8Array.from(userKey),
-      }),
+    parsedSignedDelegation: parsed_signed_delegation,
+    userKey,
   };
 }
+
+/** Run the authentication flow, including postMessage protocol, offering to authenticate
+ * using an existing anchor or creating a new anchor, etc.
+ */
+export const authenticationFlow = async (
+  connection: Connection
+): Promise<void> => {
+  const container = document.getElementById("pageContent") as HTMLElement;
+  render(html`<h1>starting authentication</h1>`, container);
+  const result = await authenticationProtocol({
+    authenticate: async (authContext) => {
+      const authSuccess = await authenticatePage(connection, authContext);
+      await recoveryWizard(authSuccess.userNumber, authSuccess.connection);
+      return authSuccess;
+    },
+    onInvalidOrigin: (result) =>
+      displayError({
+        title: "Invalid Derivation Origin",
+        message: `"${result.authContext.authRequest.derivationOrigin}" is not a valid derivation origin for "${result.authContext.requestOrigin}"`,
+        detail: result.message,
+        primaryButton: "Continue",
+      }),
+
+    onProgress: (status) => {
+      switch (status) {
+        case "waiting":
+          render(
+            html`<h1>waiting for authentication data from dapp...</h1>`,
```

Comment:
I would call it `service` instead of `dapp`. This is also what we had on the authentication page previously: "connect to service"

Reply:
Makes sense!

Reply:
Done